File: app/src/main/java/kr/mash_up/seoulmaps/base/BaseActivity.java
package kr.mash_up.seoulmaps.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.UiThread;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity {
    private Unbinder unbinder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());

        unbinder = ButterKnife.bind(this);

        initView();
    }


    @LayoutRes
    public abstract int getLayoutId();

    @UiThread
    public abstract void initView();
    

}

File: app/src/main/java/kr/mash_up/seoulmaps/base/BaseActivity.kt
package kr.mash_up.seoulmaps.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.annotation.UiThread
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())

        initView()
    }

    @LayoutRes
    abstract fun getLayoutId(): Int

    @UiThread
    abstract fun initView()
}
